### DISCUSSION TITLE
Dynamically loading locales & default locale

### DISCUSSION THREAD

**Discussion by MoonSoD** [vinissimus/next-translate#652] [Ideas]
I've been experimenting with next-translate for some days now to dynamically load available locales from our ecommerce API, so clients can add/remove their own languages. The `loadLocaleFrom` property in the config is handy but the static `locales` and `defaultLocale` properties defeat it's whole purpose IMO. I've managed to hack my way through with a custom app wrapper and loading namespaces which ofc resulted in locale routes not working. Didn't want them anyway, just used a cookie for language but **I would really like to see dynamic loading of locales out of the box, like fetching them from an API just like the locales themselves.** Very certain that many people would appreciate that. I have a feeling that this has more to do with nextjs than with the plugin itself looking at the config but didn't come that far with my research. Please let me know some of your opinions on this 😊.

**Comment by aralroca** [comment]
@MoonSoD Yes, `locales` and `defaultLocale` is part of the i18n routing of Next.js, not from this library. However, you can also change the language with a cookie they created themselves. Which cookie did you use? `NEXT_LOCALE`?  More info about this cookie:

- https://nextjs.org/docs/advanced-features/i18n-routing#leveraging-the-next_locale-cookie
- https://github.com/vinissimus/next-translate/tree/1.0.7#10-how-to-save-the-user-defined-language

Another option we implemented in Next-translate [1.0.8-canary.1](https://github.com/vinissimus/next-translate/releases/tag/1.0.8-canary.1) is the availability from the server to force a language change rewriting `req.locale`.
